Content.Shared/ComponentLocations.cs

```diff
@@ -0,0 +1,510 @@
+            ("SuitSensor", ComponentLocation.Server),
+            ("SuspicionRole", ComponentLocation.Shared),
+            ("TabletopDraggable", ComponentLocation.Shared),
+            ("TabletopGame", ComponentLocation.Server),
+            ("TabletopGamer", ComponentLocation.Server),
+            ("Tag", ComponentLocation.Shared),
+            ("TechnologyDatabase", ComponentLocation.Shared),
+            ("Telecrystal", ComponentLocation.Server),
+            ("Temperature", ComponentLocation.Server),
+            ("TemperatureProtection", ComponentLocation.Server),
+            ("ThermalRegulator", ComponentLocation.Server),
+            ("Thirst", ComponentLocation.Shared),
+            ("ThrownItem", ComponentLocation.Shared),
+            ("Thruster", ComponentLocation.Server),
+            ("TileFrictionModifier", ComponentLocation.Shared),
+            ("TilePrying", ComponentLocation.Server),
+            ("TimedSpawner", ComponentLocation.Server),
+            ("Timer", ComponentLocation.Shared),
+            ("ToggleDoorOnTrigger", ComponentLocation.Server),
+            ("Toilet", ComponentLocation.Server),
+            ("Tool", ComponentLocation.Server),
+            ("TraitorDeathMatchRedemption", ComponentLocation.Server),
+            ("TraitorDeathMatchReliableOwnerTag", ComponentLocation.Server),
+            ("Transform", ComponentLocation.Shared),
+            ("TransformableContainer", ComponentLocation.Server),
+            ("TrayScanner", ComponentLocation.Shared),
+            ("TriggerOnCollide", ComponentLocation.Server),
+            ("TriggerOnSignalReceived", ComponentLocation.Server),
+            ("TwoWayLever", ComponentLocation.Server),
+            ("Udder", ComponentLocation.Server),
+            ("UnarmedCombat", ComponentLocation.Server),
+            ("UnpoweredFlashlight", ComponentLocation.Server),
+            ("UnsimulatedGridAtmosphere", ComponentLocation.Server),
+            ("Uplink", ComponentLocation.Server),
+            ("UseDelay", ComponentLocation.Shared),
+            ("UserInterface", ComponentLocation.Shared),
+            ("Utensil", ComponentLocation.Server),
+            ("UtilityAI", ComponentLocation.Server),
+            ("Vapor", ComponentLocation.Server),
+            ("VendingMachine", ComponentLocation.Shared),
+            ("ViewSubscriber", ComponentLocation.Server),
+            ("Visibility", ComponentLocation.Server),
+            ("VisitingMind", ComponentLocation.Server),
+            ("WarpPoint", ComponentLocation.Server),
+            ("Welder", ComponentLocation.Shared),
+            ("WelderRefinable", ComponentLocation.Server),
+            ("Wieldable", ComponentLocation.Server),
+            ("Window", ComponentLocation.Shared),
+            ("WiredNetworkConnection", ComponentLocation.Server),
+            ("WirelessNetworkConnection", ComponentLocation.Server),
+            ("Wires", ComponentLocation.Shared)
```

Comment:
If we're going to do it this way this needs to be an attribute of some sort, not a big list.

Reply:
That is not possible at runtime, because the server needs to know about things that don't exist on it, and visa versa. I'll think about how to eliminate the list, but this was intended to be a better solution, even if it may not be the best solution.